workos/directory_sync.py

```diff
@@ -56,133 +56,42 @@ def list_users(
             params["directory"] = directory
 
-        if order is not None:
-            if isinstance(order, Order):
-                params["order"] = str(order.value)
-            elif order == "asc" or order == "desc":
-                params["order"] = order
-            else:
-                raise ValueError("Parameter order must be of enum type Order")
-
         response = self.request_helper.request(
             "directory_users",
             method=REQUEST_METHOD_GET,
             params=params,
             token=workos.api_key,
         )
 
-        if "default_limit" in locals():
-            if "metadata" in response and "params" in response["metadata"]:
-                response["metadata"]["params"]["default_limit"] = default_limit
-            else:
-                response["metadata"] = {"params": {"default_limit": default_limit}}
-
-        response["metadata"] = {
-            "params": params,
-            "method": DirectorySync.list_users,
-        }
-
-        return response
-
-    def list_users_v2(
-        self,
-        directory=None,
-        group=None,
-        limit=None,
-        before=None,
-        after=None,
-        order=None,
-    ):
-        """Gets a list of provisioned Users for a Directory.
-
-        Note, either 'directory' or 'group' must be provided.
-
-        Args:
-            directory (str): Directory unique identifier.
-            group (str): Directory Group unique identifier.
-            limit (int): Maximum number of records to return.
-            before (str): Pagination cursor to receive records before a provided Directory ID.
-            after (str): Pagination cursor to receive records after a provided Directory ID.
-            order (Order): Sort records in either ascending or descending order by created_at timestamp.
-
-        Returns:
-            dict: Directory Users response from WorkOS.
-        """
-
-        if limit is None:
-            limit = RESPONSE_LIMIT
-            default_limit = True
-
-        params = {
-            "limit": limit,
-            "before": before,
-            "after": after,
-            "order": order or "desc",
-        }
-
-        if group is not None:
-            params["group"] = group
-        if directory is not None:
-            params["directory"] = directory
-
-        if order is not None:
-            if isinstance(order, Order):
-                params["order"] = str(order.value)
-            elif order == "asc" or order == "desc":
-                params["order"] = order
-            else:
-                raise ValueError("Parameter order must be of enum type Order")
-
-        response = self.request_helper.request(
-            "directory_users",
-            method=REQUEST_METHOD_GET,
-            params=params,
-            token=workos.api_key,
+        return WorkOsListResource(
+            list_method=self.list_users,
+            # TODO: Should we even bother with this validation?
```

Comment:
My current opinion is no. But want to constrain scope of this PR and fix later

Reply:
Loaded up the page before you posted this, but down to omit. Better to be consistent everywhere IMO, and I think on Friday we decided against explicit input validation like this.

Reply:
👍 Will remove in a follow-on PR